superx/information_extractors/branch_info_extractor.py

```diff
@@ -152,10 +170,26 @@ def fill_branch_table(self, xml_info_list):
         branch_list = []
 
         for branch_id, branch_name, address, sub_chain_id, city in xml_info_list:
-            branch_list.append(Branch(id=branch_id,
-                                      name=branch_name, address=address,
-                                      sub_chain_id=sub_chain_id,
-                                      city=city,
-                                      chain_id=self.current_super['chain_id']))
+            # if not in db then add it
+            if (int(branch_id), self.current_super['chain_id']) not in self.branch_unique_constraint_set: # pylint: disable=line-too-long
+                branch_list.append(Branch(id=branch_id,
+                                          name=branch_name, address=address,
+                                          sub_chain_id=sub_chain_id,
+                                          city=city,
+                                          chain_id=self.current_super['chain_id']))
+                self.branch_unique_constraint_set.add((int(branch_id), self.current_super['chain_id'])) # pylint: disable=line-too-long
 
         return branch_list
+
+    def create_branch_unique_contraint_set(self):
+        """
+        queries the DB for all rows and saves the unique contraints in a set
+        """
+        all_rows = Branch.query.all()
+        for row in all_rows:
+            self.branch_unique_constraint_set.add((row.id, row.chain_id))
```

Comment:
I commented on this for the branchprice code, it is the same here, but less relevant, because this table will never be that large.